cinema/views.py

```diff
@@ -51,8 +79,61 @@
     def get_serializer_class(self):
         if self.action == "list":
             return MovieSessionListSerializer
-
         if self.action == "retrieve":
             return MovieSessionDetailSerializer
-
         return MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+        if self.action == "list":
+            queryset = queryset.annotate(
+                tickets_available=(F("cinema_hall__rows")
+                                   * F("cinema_hall__seats_in_row")
+                                   - Count("tickets")
+                                   )
+            )
+
+            date = self.request.query_params.get("date")
+            movie = self.request.query_params.get("movie")
+
+            if date:
+                queryset = queryset.filter(show_time__date=date)
+
+            if movie:
+                queryset = queryset.filter(movie__id=movie)
+
+        return queryset.distinct()
```

Comment:
Similar to the `MovieViewSet`, in the `get_queryset` method of `MovieSessionViewSet`, ensure that the use of `distinct()` is necessary. If not required, consider removing it to improve query performance.